jumanji/environments/routing/op/generator.py

```diff
@@ -0,0 +1,167 @@
+import abc
+import chex
+import jax
+import jax.numpy as jnp
+
+import types_, constants
+
+class Generator(abc.ABC):
+    """
+    Defines the abstract 'Generator' base class. A 'Generator' is responsible
+    for generating a problem instance when the environment is reset
+    """
+
+    def __init__(self, num_nodes: int, max_length: int):
+        """Abstract class implementing the attributes 'num_nodes' and 'max_length'.
+        
+        Args:
+            num_nodes (int): the number of nodes in the problem instance.
+            max_length (int): the maximum length of the tour
+        """
+        
+        self.num_nodes = num_nodes
+        self.max_length = max_length
+        
+      
+    @abc.abstractmethod
+    def __call__(self, key: chex.PRNGKey) -> types_.State:
+        """Call method responsible for generating a new state.
+        Args:
+            key: jax random key in case stochasticity is used in the instance generation process
+            
+        Returns:
+            An 'OP' environment state.
+        """    
+        
+        
+class ConstantGenerator(Generator):
```

Comment:
I would suggest to put the functions that are in the ConstantGenerator in the Generator class, with two modifications : 
(i) replacing the line `prizes = jnp.ones(self.num_nodes + 1)` by the lines 
```
key, prize_key = jax.random.split(key)
prizes = self.define_prizes(prize_key)
```
(ii) adding an abstractmethod `define_prize`

And then defining the three generators than all inherit from this base generator and all implement the relevant concrete method `define_prize`.